src/main/java/net/sf/jabref/bst/VM.java

```diff
@@ -197,7 +203,7 @@ public void execute(BstEntry context) {
                     return;
                 }
 
-                if (o1 == o2) {
```

Comment:
The if statement on line 201 checks if one of o1 and o2 are null. If both are non-null line 211 takes care of the comparison, so the only remaining case is if both are null. (Trying to avoid Object == Object.)